src/Service.Tests/SqlTests/GraphQLPaginationTests/GraphQLPaginationTestBase.cs

```diff
@@ -932,6 +932,68 @@ public async Task TestColumnTieBreak()
             SqlTestHelper.PerformTestEqualJsonStrings(expected2, actual2.ToString());
         }
 
+        /// <summary>
+        /// Tests the GraphQL query for retrieving supported types with a specified number of rows.
+        /// </summary>
+        /// <param name="pageSize">The number of rows to retrieve in the query.</param>
+        /// <param name="fields">The fields to retrieve in the query.</param>
+        /// <param name="setupQuery">The query to setup the database for the test.</param>
+        /// <param name="cleanupQuery">The query to cleanup the database after the test.</param>
+        protected async Task TestPaginantionForGivenPageSize(int pageSize, string fields, string setupQuery, string cleanupQuery)
+        {
+            // Setup for pagination adding 100 rows to the database
+            await GetDatabaseResultAsync(setupQuery);
+
+            // Arrange
+            string graphQLQueryName = "supportedTypes";
+            string graphQLQuery = $@"{{
+                supportedTypes(first: {pageSize}, after: null) {{
+                    items {{
+                        {fields}
+                    }}
+                    hasNextPage
+                    endCursor
+                }}
+            }}";
+
+            // Act
+            JsonElement graphQLResponse = await ExecuteGraphQLRequestAsync(graphQLQuery, graphQLQueryName, isAuthenticated: true);
+
+            // Assert
+            Assert.IsNotNull(graphQLResponse);
+
+            // Check for no errors
+            Assert.IsFalse(graphQLResponse.TryGetProperty("errors", out _), "Response contains errors");
+
+            // Check the number of items
+            JsonElement items = graphQLResponse.GetProperty("items");
+            Assert.AreEqual(items.GetArrayLength() == pageSize, pageSize <= 200);
```

Comment:
why does it matter pageSize to be more <=200 when items returned are = pageSize?

Reply:
because items returned won't be equal to pageSize, when the total number of rows in the table is less than the pageSize.
for example: if pageSize==1000, but no. of rows in our table is only 20. we will only get 20 rows in response.

Reply:
can you please add a comment here as to what this is asserting?

Effectively its saying that 
When PageSize > 200, returned items length CANNOT BE = whatever the pagesize is. 
When pageSize <=200, returned items length = the pagesize

Reply:
it has nothing to do with default size. Actually based on the test data, i put 200.

When PageSize > 200, returned items length CANNOT BE = whatever the pagesize is. YES, because based on the test data number of rows in the DB < 200.

So, 200 is just an arbitrary number put there to test different fields in the result based on the number of rows in the DB.